de.metas.business/src/main/java/de/metas/pricing/service/ProductPrices.java

```diff
@@ -124,6 +127,38 @@ public static void assertMainProductPriceIsNotDuplicate(final I_M_ProductPrice p
 		getFirstOrThrowExceptionIfMoreThanOne(allMainPrices);
 	}
 
+	public static void assertUomConversionExists(final I_M_ProductPrice productPrice)
+	{
+		if (productPrice == null || !productPrice.isActive())
+		{
+			return;
+		}
+
+		if (productPrice.isInvalidPrice())
+		{
+			return;
+		}
+
+		final IProductDAO productDAO = Services.get(IProductDAO.class);
+		final org.compiere.model.I_M_Product product = productDAO.getById(productPrice.getM_Product_ID());
+
+		if (UomId.ofRepoId(product.getC_UOM_ID()).equals(UomId.ofRepoId(productPrice.getC_UOM_ID())))
+		{
+			return;
+		}
+
+		final IUOMConversionDAO uomConversionRepo = Services.get(IUOMConversionDAO.class);
+		final ProductId productId = ProductId.ofRepoId(productPrice.getM_Product_ID());
+
+		UOMConversionsMap conversionsMap = uomConversionRepo.getProductConversions(productId);
+
+		if (conversionsMap.getRateIfExists(UomId.ofRepoId(product.getC_UOM_ID()), UomId.ofRepoId(productPrice.getC_UOM_ID())).isPresent())
+		{
+			return;
+		}
+		throw new AdempiereException("UOM Conversion to the selected UOM doesn't exist").markAsUserValidationError();
```

Comment:
Create an `AD_Message` record with translations (we'll need en_US, de_DE and de_CH) and use IMsg (`Services.get(IMsg.class)`) to get a translatable string instance.
Passing hardcoded strings  generally doesn't make sense for user-errors

Reply:
fully agree